### PR TITLE
Improve and simplify field typing overloads

### LINKED ISSUE #85: False positive for argument mismatch
```
# pyright: reportGeneralTypeIssues=true
from django.db import models


class Test(models.Model):
    x = models.ForeignKey["Test"]("test.Test", models.CASCADE, null=True)
    y = models.ForeignKey("test.Test", models.CASCADE, null=True)
```

In this example, `x = ...` triggers following error in pyright:

```
  /tmp/test.py:6:69 - error: Argument of type "Literal[True]" cannot be assigned to parameter "null" of type "Literal[False]" in function "__init__"
    "Literal[True]" cannot be assigned to type "Literal[False]" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

But i don't understand why, code is correct, and typing looks alright.

https://github.com/sbdchd/django-types/blob/main/django-stubs/db/models/fields/related.pyi#L93

Any ideas? Or does this looks like error in pyright?

### LINKED ISSUE #87: Remove the overloads for choices and simplify typing
Hey @sbdchd ,

I'm thinking a lot regarding the change I've proposed. Although it is very nice, it complicated the typing of the fields a lot and it has some problems with pyright and mypy due to the fact that they require both `__init__` and `__new__` to be overloaded.

I'm thinking in opening a PR simplifying that and removing the `__init__` from everything. I think it should be enough to define just 2 `__new__` (one for the null=True other for the null=False) with the accepted arguments.

Do you agree with my point of view here?

### PR DESCRIPTION
**Pull request by bellini666**
Fix #87
Fix #85

### CODE REVIEW

**Comment by bellini666** [comment]
@sbdchd just like we talked, this sets `__new__` overload on all fields instead of using a mix of `__init__`/`__new__`, which is not a documented behaviour.

**Comment by bellini666** [comment]
Hey @sbdchd , is everything fine with this PR?

File: tests/pyright/test_fields.py [hunk 1]
```diff
@@ -121,28 +97,16 @@ class Foo(models.Model):
...
-        ),
         Result(
             type="error",
             message='Cannot assign member "char_with_choices" for type "Foo" (reportGeneralTypeIssues)',
             line=15,
             column=3,
         ),
-        Result(
-            type="error",
-            message='Cannot assign member "char_with_choices" for type "Foo" (reportGeneralTypeIssues)',
```
Comment (sbdchd) [inline tests/pyright/test_fields.py:142]: curious if these changes are expected, seems were losing some strictness with the changes

**Comment by sbdchd** [comment]
Sorry for the delay! I'm always super hesitant with big changes because I want to avoid breaking downstream consumers.

Can we add some tests to show what we get from this change?

File: tests/pyright/test_fields.py (hunk 1 above)
Comment (bellini666) [reply tests/pyright/test_fields.py:142]: We are actually just loosing the strictness that I previously added for the "Literal" capturing of the `choices` attributes. Other than that, the change should not have removed any strictness

**Comment by bellini666** [comment]
@sbdchd np, I understand your hesitation! :)

The change that I'm removing is the one that I previously forced and even ended up not even using it because I prefer to use my https://github.com/bellini666/django-choices-field to properly type the field with enums.

I know see that I overcomplicated the typing, which makes maintaining and improving it a lot harder, and also produced issues with the fact that we have both a `__init__` and a `__new__` (the alternative was to have 4 `__new__` for literally each field, which is also really bad in terms of maintenance)

Other than that, everything should be literally the same.

obs. If someone still wants to type the field's choices, he can do something like:

```python
class SomeModel(models.Model):
    choices = Literal["foo", "bar", "bin"]
    some_field = models.CharField[choices]()
```

(but I highly doubt that anyone was depending on this)

> Can we add some tests to show what we get from this change?

Hrm, I could do that, but don't know actually what to write for tests because we are not getting anything from this change other than a very nice and needed simplification of the fields' overloads.

It fixes the #85, but it is actually a side effect of the fact that we are not doing the wrong thing of having both `__init__` (with `self` being typed) and `__new__` at the same classes.

**Comment by sbdchd** [comment]
Thank you for the explanation and all your hard work, looks great!

**Pull request merged**